### PR TITLE
Updated code-coverage in rootfs_image.go

### PR DESCRIPTION
**Pull request by oleorhagen**
Signed-off-by: Ole Petter <ole.orhagen@cfengine.com>

### CODE REVIEW

File: artifact/tar_writer_stream.go [hunk 1]
```diff
@@ -43,6 +43,9 @@ func NewTarWriterStream(w *tar.Writer) *StreamArchiver {
 }
 
 func (str *StreamArchiver) Write(data []byte, archivePath string) error {
+	if str.Writer == nil {
+		return errors.New("arch: Can not write to empty tar-writer")
```
Comment (oleorhagen) [inline artifact/tar_writer_stream.go]: I think it is better to return an error than to panic on an empty tar writer.

File: handlers/common.go [hunk 2]
```diff
@@ -184,6 +184,9 @@ func match(pattern, name string) bool {
 }
 
 func writeFiles(tw *tar.Writer, updFiles []string, dir string) error {
+	if tw == nil {
```
Comment (oleorhagen) [inline handlers/common.go]: same

File: handlers/rootfs_image_test.go [hunk 3]
```diff
@@ -148,3 +150,183 @@ func TestRootfsReadHeader(t *testing.T) {
...
+			rfs:  NewRootfsV2(""),
+			args: ComposeHeaderArgs{},
+			err:  errors.New("writer: tar-writer is nil"),
+		},
+		"version 3 - no tar writer": {
+			rfs:  NewRootfsV3(""),
+			args: ComposeHeaderArgs{},
+			err:  errors.New("ComposeHeader: Payload: can not tar type-info: arch: Can not write to empty tar-writer"),
+		},
+		"version 1 - success": { // TODO - should this succeed with no update files?
```
Comment (oleorhagen) [inline handlers/rootfs_image_test.go]: Not sure. Which is correct?

File: handlers/rootfs_image_test.go [hunk 4]
```diff
@@ -148,3 +150,183 @@ func TestRootfsReadHeader(t *testing.T) {
...
+			if test.verifyFunc != nil {
+				test.verifyFunc(test.args)
+			}
+		}
+	}
+}
+
+func TestNewAugmentedInstance(t *testing.T) {
+	rfs := NewRootfsInstaller()
+	inst, err := rfs.NewAugmentedInstance(NewRootfsV2(""))
```
Comment (oleorhagen) [inline handlers/rootfs_image_test.go]: The rest are tests for the setters and getters of the udpates. A fair bit of code. Not really sure how helpful they are though, except increasing the coverage. It did find the nil pointer tar writer panic though.

File: handlers/rootfs_image_test.go (hunk 3 above)
Comment (kacf) [reply handlers/rootfs_image_test.go]: I'm not sure either, but I wouldn't worry too much about it. We have completely disabled v1 support on the reading side, in anticipation of removing it altogether.

File: handlers/rootfs_image_test.go (hunk 4 above)
Comment (kacf) [reply handlers/rootfs_image_test.go]: Perhaps not useful right now, but they will be in the future, when we switch rootfs updates over to augmented updates (which was the original plan). For now it's a future investment.

File: artifact/tar_writer_stream.go (hunk 1 above)
Comment (kacf) [reply artifact/tar_writer_stream.go]: :+1:

File: handlers/common.go (hunk 2 above)
Comment (kacf) [reply handlers/common.go]: :+1:

**Pull request merged**